common/src/main/java/com/flipkart/varadhi/common/events/EntityEventListener.java

```diff
@@ -23,5 +23,5 @@ public interface EntityEventListener {
      * @param event the event containing information about the entity change
      * @throws IllegalStateException if the listener is in a state where it cannot process events
      */
-    void onChange(EntityEvent<?> event);
+    void onChange(EntityEvent<T> event);
```

Comment:
due to this change, now the DefaultMetaStoreChangeListener will throw warning.

There we are using it as 
`private final EntityEventListener listener;`
<img width="675" alt="Screenshot 2025-04-14 at 5 10 11 PM" src="https://github.com/user-attachments/assets/313ad974-940a-42fc-9d1b-1b200ffb644b" />


Ad discussed, it probably should be `void onChange(EntityEvent<? extends T> event);`